### PR TITLE
Fix locale tests for the stricter `parse`

### PR DESCRIPTION
**Pull request by leshakoss**
Recently we merged a PR #821 that makes `parse` a bit stricter (forbids any remaining after parsing input), which uncovered a number of typos and problems in locales and their tests. The following locales were affected: de, es, fr, nb, nl, pt-BR, ru, sv, th, uk, zh-CN

### CODE REVIEW

**Comment by leshakoss** [comment]
@date-fns/i18n-zh-cn I had to make considerable changes to zh-CN locale to fix the broken tests: https://github.com/date-fns/date-fns/pull/823/files#diff-7da1c3aa1894e9c7acf037b892b0ad48L23 I used mostly google translate and [CLDR chart](https://www.unicode.org/cldr/charts/32/summary/zh.html). Could you please check it and tell me what did I get wrong?

@date-fns/i18n-nl I swapped around some values for months and weekdays in Dutch locale: https://github.com/date-fns/date-fns/pull/823/files#diff-1e47917961857bc076b8fe0affc8c426L13 to make them more like the [CLDR chart](https://www.unicode.org/cldr/charts/32/summary/nl.html). Please check them out

**Comment by leshakoss** [comment]
I am going to merge it now in order to release the [this fix](https://github.com/date-fns/date-fns/pull/821/files/f1e32093ec85b96ad85a036f41684be3b29a03d5#diff-633a67ff28ab244d01814c402674e71bL520) as soon as possible, but please still take a look at the changes in this PR.

File: src/locale/zh-CN/_lib/match/index.js [hunk 1]
```diff
@@ -43,19 +43,18 @@ var parseDayPatterns = {
...
-  narrow: /^[早中午晚上下凌晨夜]/i,
-  any: /^[早中午晚上下凌晨夜]/i
+  any: /^(上午|下午|午夜|[中正]午|早上|下午|晚上?|凌晨)/i
 }
 var parseDayPeriodPatterns = {
   any: {
-    am: /[上早凌]/i,
-    pm: /[下晚夜]/i,
-    midnight: /[夜]/i,
+    am: /^上午/i,
```
Comment (cubicwork) [inline src/locale/zh-CN/_lib/match/index.js]: 'a.m.' includes '上午', '早上' and '凌晨'

File: src/locale/zh-CN/_lib/match/index.js [hunk 2]
```diff
@@ -43,19 +43,18 @@ var parseDayPatterns = {
...
+    pm: /^下午/i,
```
Comment (cubicwork) [inline src/locale/zh-CN/_lib/match/index.js]: 'p.m.' includes '下午', '傍晚', '晚上' and '深夜'

File: src/locale/zh-CN/_lib/match/index.js [hunk 3]
```diff
@@ -43,19 +43,18 @@ var parseDayPatterns = {
...
+    midnight: /^午夜/i,
     noon: /^[中正]午/i,
-    morning: /早/i,
-    afternoon: /下午/i,
-    evening: /晚/i,
-    night: /夜/i
+    morning: /^早上/i,
+    afternoon: /^下午/i,
+    evening: /^晚/i,
+    night: /^凌晨/i
```
Comment (cubicwork) [inline src/locale/zh-CN/_lib/match/index.js]: 'night' includes '深夜' and '晚上'

File: src/locale/zh-CN/_lib/match/index.js [hunk 4]
```diff
@@ -43,19 +43,18 @@ var parseDayPatterns = {
...
+    pm: /^下午/i,
+    midnight: /^午夜/i,
     noon: /^[中正]午/i,
-    morning: /早/i,
-    afternoon: /下午/i,
-    evening: /晚/i,
-    night: /夜/i
+    morning: /^早上/i,
+    afternoon: /^下午/i,
+    evening: /^晚/i,
```
Comment (cubicwork) [inline src/locale/zh-CN/_lib/match/index.js]: 'evening' includes '傍晚' and '晚上'

File: src/locale/nl/_lib/localize/index.js [hunk 5]
```diff
@@ -13,13 +13,14 @@ var quarterValues = {
...
-  narrow: ['jan', 'feb', 'mrt', 'apr', 'mei', 'jun', 'jul', 'aug', 'sep', 'okt', 'nov', 'dec'],
+  narrow: ['J', 'F', 'M', 'A', 'M', 'J', 'J', 'A', 'S', 'O', 'N', 'D'],
+  abbreviated: ['jan.', 'feb.', 'mrt.', 'apr.', 'mei.', 'jun.', 'jul.', 'aug.', 'sep.', 'okt.', 'nov.', 'dec.'],
   wide: ['januari', 'februari', 'maart', 'april', 'mei', 'juni', 'juli', 'augustus', 'september', 'oktober', 'november', 'december']
 }
 
 var dayValues = {
-  narrow: ['zo', 'ma', 'di', 'wo', 'do', 'vr', 'za'],
-  short: ['zon', 'maa', 'din', 'woe', 'don', 'vri', 'zat'],
+  narrow: ['Z', 'M', 'D', 'W', 'D', 'V', 'Z'],
```
[the remark this answers is not in the document]
Comment (leshakoss) [reply src/locale/nl/_lib/localize/index.js]: @curry684 The calendar use case is good enough for me. After all, date-fns is sometimes used under the hood in actual calendars (for example, [here](https://github.com/clauderic/react-infinite-calendar)).

I thought it didn't make sense to make `short` and `abbreviated` formats of weekdays the same (`maa`, `din`, `woe` etc) when there is a one-letter format uncovered. But the main reasoning is that this conforms to [the standard](https://www.unicode.org/cldr/charts/32/summary/nl.html#1921) which we try to use when there is some ambiguity.

As for ambiguous parsing of one-letter months and weekdays, the priority is to 1) follow the CLDR standard, and 2) cover any use case with `format` and then mirror them in `parse` even when unambiguous parsing is not possible. [The situation is the same in en-US locale](https://github.com/date-fns/date-fns/blob/master/src/locale/en-US/_lib/localize/index.js#L20).

This is still up to discussion, but for now I think that usage of this format in calendars along with conforming to the standard is convincing enough to me.

File: src/locale/zh-CN/_lib/match/index.js (hunk 3 above)
Comment (leshakoss) [reply src/locale/zh-CN/_lib/match/index.js]: > '深夜' means 2200-0100

Ouch. Sounds like we'll probably have to change API a bit for that

**Comment by Lakston** [comment]
I had issues with the way I took care of the parsing for the French locale so it will be a way for me to make a second pass on it. I'll probably have some questions / need help with some things though, regex is not my forte :)

**Pull request merged**